### PR TITLE
Adds npc_chat_api module to demonstrate Smart NPC use-case

### PR DESCRIPTION
**Pull request by zmerlynn**
* Adds a basic "Smart NPC" implemented by RAG using embedding/prediction from Vertex (currently) and Cloud Spanner. See README for more details.

* Adds npc-chat Spanner instance/database for npc_chat_api module.

* Adds spanner permissions to service account.

### CODE REVIEW

File: genai/api/npc_chat_api/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+> [!NOTE]
+> Work in progress!
+
+The NPC Chat API demonstrates using [RAG](https://www.promptingguide.ai/techniques/rag) to create a 
+smart NPC for your game, enriched by world knowledge, NPC-specific knowledge, and secondhand
+knowledge from chat.
+
+To try it out:
+
```
Comment (igooch) [inline genai/api/npc_chat_api/README.md:11]: I'd make a note that you need to first set up the Quickstart cluster and run the `sed` command.

Comment (zmerlynn) [reply genai/api/npc_chat_api/README.md:11]: I think it's safe to assume the reader has created the Quickstart cluster if they're in a deep directory here.

The `sed` thing will go away with https://github.com/googleforgames/GenAI-quickstart/pull/13

**Review by zaratsian — APPROVED** [review]
Tested successfully on my end, LGTM.

**Review by smithpatrick12 — APPROVED** [review]
Reviewed - LGTM

**Pull request merged**